### PR TITLE
add pallet-parentchain 

### LINKED ISSUE #1182: add parentchain pallet to pallets
### Context
In tee-worker, it depends on `pallet-parentchain` from `integritee-network/pallets`.

### Task
We need to add `pallet-parentchain` to our pallets dir, like `teerex`.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by zTgx**
In `tee-worker`, it depends on `teerex-primitives` & `pallet-parentchain`.

- [x] replace `teerex-primitives` dependence in Cargo.toml
- [x] add `pallet-parentchain` to `pallets`  & update Cargo.toml

### CODE REVIEW

File: tee-worker/Cargo.toml [hunk 1]
```diff
@@ -88,27 +88,3 @@ sgx_tstd = { version = "1.1.6", git = "https://github.com/apache/incubator-teacl
 sgx_types = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
 sgx_ucrypto = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
 sgx_urts = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
-
```
Comment (BillyWooo) [inline tee-worker/Cargo.toml]: I would suggest not to touch these commented lines, until the upstream remove it.

File: tee-worker/enclave-runtime/Cargo.toml [hunk 2]
```diff
@@ -173,14 +173,3 @@ sgx_tseal = { version = "1.1.6", git = "https://github.com/apache/incubator-teac
 sgx_tstd = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
 sgx_tunittest = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
 sgx_types = { version = "1.1.6", git = "https://github.com/apache/incubator-teaclave-sgx-sdk", branch = "master" }
-
```
Comment (BillyWooo) [inline tee-worker/enclave-runtime/Cargo.toml]: Same as above. Not touch it, until upstream remove it.

File: tee-worker/Cargo.toml (hunk 1 above)
Comment (zTgx) [reply tee-worker/Cargo.toml]: sounds reasonable, will recover it.

File: tee-worker/enclave-runtime/Cargo.toml (hunk 2 above)
Comment (zTgx) [reply tee-worker/enclave-runtime/Cargo.toml]: recovered

**Review by BillyWooo — APPROVED** [review]
LGTM

**Pull request merged**